### DISCUSSION TITLE
Error during checkpoint removal: [internal error: bitmap 'virtnbdbackup.1' not found in backing chain of 'vda'] I can't remove checkpoints.

### DISCUSSION THREAD

**Discussion by linuxgokan** [abbbi/virtnbdbackup#223] [General]
Hello

I can't remove checkpoints. I will be backing up to a new NAS device.

virsh checkpoint-list vm1 
 Name              Creation Time
----------------------------------------------
 virtnbdbackup.0   2024-09-19 20:37:37 +0300
 virtnbdbackup.1   2024-09-19 21:05:40 +0300
 virtnbdbackup.2   2024-09-19 21:08:31 +0300
 virtnbdbackup.3   2024-09-20 08:53:18 +0300

[2024-09-27 14:40:21] ERROR root checkpoint - delete [main]:  Error during checkpoint removal: [internal error: bitmap 'virtnbdbackup.1' not found in backing chain of 'vda']
[2024-09-27 14:40:21] ERROR root virtnbdbackup - main [main]:  Failed to remove checkpoint.

**Comment by linuxgokan** [comment]
I removed all checkpoints and opened them again but it still says there is a checkpoint.

cd /var/lib/libvirt/qemu/checkpoint/vm1
ls -l
virtnbdbackup.0.xml
virtnbdbackup.1.xml
virtnbdbackup.2.xml
virtnbdbackup.3.xml
rm -rf *

**Comment by abbbi** [comment]
remove checkpoint using virsh utility, with --metadata option.
Removing files will not delete checkpoints at libvirt api.

https://github.com/abbbi/virtnbdbackup?tab=readme-ov-file#backup-fails-with-error-during-checkpoint-removal-internal-error-bitmap-xx-not-found-in-backing-chain-of-xx
